pkg/image/validation/kubernetes_test.go

```diff
@@ -1079,3 +1070,691 @@ func TestValidateAdditionalArtifacts(t *testing.T) {
+			},
+			ExpectedFailedMessages: []string{
+				"Non-unicast cluster API address (FF01::1) for field 'apiVIP6' is invalid.",
+			},
+		},
+		`link-local v4 VIP`: {
+			K8s: image.Kubernetes{
+				Network: image.Network{
+					APIHost: "host.com",
+					APIVIP4: "169.254.1.1",
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"Non-unicast cluster API address (169.254.1.1) for field 'apiVIP' is invalid.",
+			},
+		},
+		`link-local v6 VIP`: {
+			K8s: image.Kubernetes{
+				Network: image.Network{
+					APIHost: "host.com",
+					APIVIP6: "FE80::1",
+				},
+			},
+			ExpectedFailedMessages: []string{
+				"Non-unicast cluster API address (FE80::1) for field 'apiVIP6' is invalid.",
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			k := test.K8s
+			failures := validateNetwork(&k)
+			assert.Len(t, failures, len(test.ExpectedFailedMessages))
+
+			var foundMessages []string
+			for _, foundValidation := range failures {
+				foundMessages = append(foundMessages, foundValidation.UserMessage)
+			}
+
+			for _, expectedMessage := range test.ExpectedFailedMessages {
+				assert.Contains(t, foundMessages, expectedMessage)
+			}
+
+		})
+	}
+}
+
+func TestValidateConfigInvalidServerConfigNotConfigured(t *testing.T) {
+	k8s := image.Kubernetes{Network: image.Network{
+		APIVIP4: "192.168.1.1",
+		APIVIP6: "fd12:3456:789a::21",
+	}}
+
+	failures := validateNetworkingConfig(&k8s, "fake-path")
+
+	assert.Len(t, failures, 1)
+
+	var foundMessages []string
+	for _, foundValidation := range failures {
+		foundMessages = append(foundMessages, foundValidation.UserMessage)
+	}
+
+	assert.Contains(t, foundMessages, "Kubernetes server config could not be found at 'fake-path'; dual-stack configuration requires a valid cluster-cidr and service-cidr.")
+}
+
+func TestValidateConfigValidAPIVIPNotConfigured(t *testing.T) {
+	k8s := image.Kubernetes{}
+
+	failures := validateNetworkingConfig(&k8s, "")
+	assert.Len(t, failures, 0)
+}
+
+func TestValidateConfigValidIPv4Prio(t *testing.T) {
+	k8s := image.Kubernetes{Network: image.Network{
+		APIVIP4: "192.168.1.1",
+		APIVIP6: "fd12:3456:789a::21",
+	}}
+
+	configDir, err := os.MkdirTemp("", "eib-config-")
+	require.NoError(t, err)
+
+	defer func() {
+		assert.NoError(t, os.RemoveAll(configDir))
+	}()
+
+	serverConfigDir := filepath.Join(configDir, "kubernetes", "config")
+	require.NoError(t, os.MkdirAll(serverConfigDir, os.ModePerm))
+
+	serverConfig := map[string]any{
+		"cluster-cidr": "10.42.0.0/16,fd12:3456:789b::/48",
+		"service-cidr": "10.43.0.0/16,fd12:3456:789c::/112",
+	}
+
+	b, err := yaml.Marshal(serverConfig)
+	require.NoError(t, err)
+
+	configFile := filepath.Join(serverConfigDir, "server.yaml")
+	require.NoError(t, os.WriteFile(configFile, []byte(b), 0o600))
+
+	failures := validateNetworkingConfig(&k8s, configFile)
+
+	assert.Len(t, failures, 0)
+}
+
+func mockValidateConfig(k8s *image.Kubernetes, serverConfig map[string]any) []FailedValidation {
```

Comment:
Mocking means to provide fake data as a result and we do calls to the actual functions we want to test within this function.

We cannot rely on the current implementation to only include the calls to `validateNodeIP` and `validateCIDRConfig` in the future. Removing either of those or adding anything else will render the  `TestValidateConfig` "useless". Let's drop it since we already have separate test tables covering both of these functions separately.